src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -441,10 +441,14 @@ protected Query doToQuery(QueryShardContext context) {
             }
             spaceType.validateVector(byteVector);
         } else if (VectorDataType.BYTE == vectorDataType) {
-            byteVector = new byte[vector.length];
+            if (KNNEngine.LUCENE == knnEngine) {
```

Comment:
This is quite confusing. Can we just use byte here same as lucene and only convert it back to float right before calling JNI?

Reply:
refactored it with a switch case